grammar.js

```diff
@@ -535,6 +538,103 @@ module.exports = grammar({
 
     dotted_name: $ => sep1($.identifier, '.'),
 
+    // Match cases
+
+    case_pattern: $ => choice(
+      $.case_as_pattern,
+      $.case_or_pattern,
+    ),
+
+    case_as_pattern: $ => seq(
+      field('or_pattern', $.case_or_pattern), 'as', field('identifier', $.identifier)),
+
+    case_or_pattern: $ => seq(
+      $.case_closed_pattern, repeat(seq('|', $.case_closed_pattern))),
+
+
+    case_closed_pattern: $ => choice(
+      $.case_literal_pattern,
+      $.dotted_name,
+      $.case_wildcard_pattern,
+      $.case_group_pattern,
+      $.case_sequence_pattern,
+      $.case_mapping_pattern,
+      $.case_class_pattern,
+    ),
+
+    case_literal_pattern: $ => choice(
+      $.string,
+      $.concatenated_string,
+      $.case_literal_pattern_complex_number,
+      $.case_literal_integer_or_float,
+      $.true,
+      $.false,
+      $.none,
+    ),
+
+    case_literal_pattern_complex_number: $ => seq(
+      field('real_component', $.case_literal_integer_or_float),
+      field('sign', choice('+', '-')),
+      field('imaginary_component', $.case_literal_integer_or_float),
```

Comment:
Now here, the first and last field *could* make sense, as they're the same rule, however, since there's no repeats or optional usages to make it ambiguous, query matching can easily be done by just using the anchor or adding the +/- sign here.

So, still unnecessary